std/experimental/allocator/building_blocks/scoped_allocator.d

```diff
@@ -89,6 +89,8 @@ struct ScopedAllocator(ParentAllocator)
         toInsert.prev = null;
         toInsert.next = root;
         toInsert.length = n;
+        assert(!root || !root.prev);
```

Comment:
On top of the doubly-linked list actually ;-)
As we are at this, I'd suggest to add "Please, file an issue" or something like that, because if the assertion triggers, it's not a precondition gone wrong from the user, but rather an implementation bug.